codice/ddf
# DDF-04343 As a user, I would like to enter Degrees, Minutes, Seconds coordinates into the gazetteer search box so I can pan to a certain location quickly

#### What does this PR do?
Adds a dmsProcessor which the search gazetteer can use to pan/zoom to a certain location.
#### Who is reviewing it? 

@Lambeaux 
@dimitry-sherman 
@BernardIgiri 
#### Select relevant component teams: 

#### Ask 2 committers to review/merge the PR and tag them here.

@jrnorth
@millerw8 

#### How should this be tested?
1. Turn off the online gazetteer.
2. Test various DMS coordinates with the search gazetteer, alone and in combination:

```
12°34'56.789"N 123°45'6"E
33 3 9S 08 45 42.44444W
90 0 0N 180 0 0E
^88$5grg4S 1 3...24E (88°05'04"S 001°03'24"E)
```
Vary the casing (upper and lower case for cardinal direction). Ensure clicking the DMS suggestion from the search box pans you to the expected location, and that invalid coordinates are not suggested. Right clicking on the map and copying the DMS coordinate is one way to ensure you end up where you expect. 

Use the following converter if needed to check accuracy: http://www.earthpoint.us/Convert.aspx

#### Any background context you want to provide?
#### What are the relevant tickets?
https://github.com/codice/ddf/issues/4343

#### Screenshots

![image](https://user-images.githubusercontent.com/4467636/52755613-c3178880-2fbb-11e9-8187-f8d647b1336d.png)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
